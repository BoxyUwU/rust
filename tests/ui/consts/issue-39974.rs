const LENGTH: f64 = 2;
//~^ ERROR: mismatched types

struct Thing {
    f: [[f64; 2]; LENGTH],
    //~^ ERROR: the constant `[const error]` is not of type `usize`
}

fn main() {
    let _t = Thing { f: [[0.0, 0.0], [0.0, 0.0], [0.0, 0.0], [0.0, 0.0]] };
}
