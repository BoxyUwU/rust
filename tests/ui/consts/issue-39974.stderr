error[E0308]: mismatched types
  --> $DIR/issue-39974.rs:1:21
   |
LL | const LENGTH: f64 = 2;
   |                     ^
   |                     |
   |                     expected `f64`, found integer
   |                     help: use a float literal: `2.0`

error: the constant `[const error]` is not of type `usize`
  --> $DIR/issue-39974.rs:5:8
   |
LL |     f: [[f64; 2]; LENGTH],
   |        ^^^^^^^^^^^^^^^^^^ expected `usize`, found `f64`

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0308`.
